timemachine/cpp/src/bd_exchange_move.cu

```diff
@@ -217,21 +218,39 @@ void BDExchangeMove<RealType>::move(
             d_lse_exp_sum_after_.data,
             stream);
 
-        k_attempt_exchange_move<RealType><<<1, 1, 0, stream>>>(
-            N,
-            d_mh_noise_.data + (step * batch_size_),
+        k_select_first_valid_move<RealType><<<1, min(512, batch_size_), 0, stream>>>(
+            num_proposals_per_move_,
+            num_target_mols_,
+            batch_size_,
+            d_noise_offset_.data,
+            d_samples_.data,
             d_lse_max_before_.data,
             d_lse_exp_sum_before_.data,
             d_lse_max_after_.data,
             d_lse_exp_sum_after_.data,
-            d_target_mol_offsets_.data,
+            d_mh_noise_.data,
+            d_selected_sample_.data);
+        gpuErrchk(cudaPeekAtLastError());
+
+        k_accepted_exchange_move<<<1, 1, 0, stream>>>(
```

Comment:
naming, why past-tense `k_accepted` vs present-tense `k_accept`? Did the attempt *already* happen by this point?

Reply:
Yes, the acceptance is done inside of the `k_select_first_valid_move` which should probably be `k_accept_first_valid_move`

Reply:
Renamed `k_select_first_valid_move` in https://github.com/proteneer/timemachine/pull/1274/commits/debf9b2c97397497ae5e9b10ab16b325326b4d1f